modules/deploying-models-using-multiple-gpu-nodes.adoc

```diff
@@ -0,0 +1,266 @@
+
+* `workerSpec.tensorParallelSize`: Determines how many GPUs are used per node. The GPU type count in both the head and worker node deployment resources is updated automatically. Ensure that the value of `workerSpec.tensorParallelSize` is at least `1`.
+* `workerSpec.pipelineParallelSize`: Determines how many nodes are involved in the deployment. This variable represents the total number of nodes, including both the head and worker nodes. Ensure that the value of `workerSpec.pipelineParallelSize` is at least `2`.
+
+== Serving a vLLM model using multiple nodes
+
+This procedure describes how to serve a vLLM model using a multi-node setup. 
+
+NOTE: If you have already configured a persistent volume claim (PVC) and uploaded a model for deployment, you can skip the first 2 steps.
+
+.Procedure
+
+. In a terminal window, if you are not already logged in to your {openshift-platform} cluster as a cluster administrator, log in to the OpenShift CLI as shown in the following example:
++
+[source]
+----
+$ oc login <openshift_cluster_url> -u <admin_username> -p <password>
+----
++
+. From the namespace where you would like to deploy the model, create a PVC for model storage and provide the name of your storage class. The storage class must be file storage.
++
+[source]
+----
+kubectl apply -f - <<EOF
+apiVersion: v1
+kind: PersistentVolumeClaim
+metadata:
+  name: llama-3-8b-pvc
+spec:
+  accessModes:
+    - ReadWriteMany
+  volumeMode: Filesystem
+  resources:
+    requests:
+      storage: 50Gi
+  storageClassName: __<fileStorageClassName>__
+EOF
+----
++
+. Download the model by exporting your token as an environment variable. In this example, `HF_TEST_TOKEN` is the exported token:
++
+[source]
+----
+export HF_TEST_TOKEN=__<token>__
+export MODEL=meta-llama/Meta-Llama-3-8B-Instruct
+
+curl -o download-model-to-pvc.yaml https://raw.githubusercontent.com/kserve/website/a5a8661de3c3bd55b1faa1acf9125d05b10bfefd/docs/modelserving/v1beta1/llm/huggingface/multi-node/download-model-to-pvc.yaml
+envsubst < download-model-to-pvc.yaml | kubectl create -f -
+----
++
+
+ifndef::upstream[]
+. From {productname-long}, create a custom model serving runtime in your data science project. For more information, see link:{rhoaidocshome}{default-format-url}serving_models/serving-large-models_serving-large-models#adding-a-custom-model-serving-runtime-for-the-single-model-serving-platform_serving-large-models[Adding a custom model-serving runtime].
+endif::[]
+ifdef::upstream[]
+. From {productname-long}, create a custom model serving runtime in your data science project. For more information, see link:{odhdocshome}/serving-models/#adding-a-custom-model-serving-runtime-for-the-multi-model-serving-platform_model-serving[Adding a custom model-serving runtime].
+endif::[]
+
+. Modify the serving runtime with the path to your `vllm-multinode-runtime.yaml` file.
```

Comment:
is this needed? the flow is to use a template

Reply:
@Jooho What do you think?

Reply:
The template is not shown up in the dashboard so we would provide a yaml file. by the way, where is `vllm-multinode-runtime.yaml`file path? 

I think we can provide a way to create this template in user namespace for who can not access to redhat-ods-applications namespace. like the following:

```
curl -L https://raw.githubusercontent.com/opendatahub-io/odh-model-controller/refs/heads/main/config/runtimes/vllm-multinode-template.yaml -o /tmp/vllm-multinode-template.yaml

cat /tmp/vllm-multinode-template.yaml |sed "s+image: .*$+image: quay.io/modh/vllm@sha256:ce5a8e5fc91442229ea4f54ea251cf65f10d383021138f8f0d55dcc5c3eb038b+g"  |oc apply -f -
```

Reply:
the template is available on the cluster and it was decided to let only admins use it so i think we should go with `oc process`
i also think we should not point to the template in the repo as it may be updated but not tested yet in d/s.

Reply:
Just so I'm clear, should we leave this step as-is or are there changes I should make here?

Reply:
@rnetser This is the one that tarun brought up last time. have you tested to create the servingruntime through template with normal user?

Reply:
A normal user cannot access the template and we agreed that only an admin can do it.